### DISCUSSION TITLE
Cross compiling for arm

### DISCUSSION THREAD

**Discussion by Rikorose** [leptos-rs/leptos#2803] [Q&A]
I want to cross compile from linux x64 to aarch64. On the x64 machine I put `bin-target-triple="aarch64-unknown-linux-gnu"` and compiled frontend and server using: `LEPTOS_BIN_CARGO_COMMAND=cross cargo leptos build --release`. This works so far.

I am not sure how to correctly execute the resulting binary on the aarch64 machine. I tried to directly call the executable: `./target/aarch64-unknown-linux-gnu/release/leptos-binary-test`:
```
It looks like you're trying to compile Leptos without the LEPTOS_OUTPUT_NAME environment variable being set. There are two options
 1. cargo-leptos is not being used, but get_configuration() is being passed None. This needs to be changed to Some("Cargo.toml")
 2. You are compiling Leptos without LEPTOS_OUTPUT_NAME being set with cargo-leptos. This shouldn't be possible!
listening on http://127.0.0.1:3000
```

It seems, that some config is not correctly passed to the binary also since the port is the default port (which I modified).

Or is there some way to run the cross compiled binary with cargo leptos?

**Comment by Itsskiip** [comment]
I ran into the same issue, I solved it by changing ```get_configuration(None)``` to ```get_configuration(Some("Cargo.toml"))``` near the root of the program as described in the error message, then also copying ```Cargo.toml``` to the same directory as the file.

**Comment by bochaco** [comment]
I get the same message at runtime even that when cross-compiling with `cross` I set `LEPTOS_OUTPUT_NAME` env var myself.

Plus, it also builds the frontend to fetch the wasm file with `_bg.wasm` suffix.

**Reply by bochaco** [reply]
The issue seems to be that within `cross` environment is not seeing the `LEPTOS_OUTPUT_NAME` env var I was setting, I just fixed it by adding the following to my `Cargo.toml` file:
```
[package.metadata.cross.build.env]
passthrough = ["LEPTOS_OUTPUT_NAME=<desired-output-name>"]
```
For some reason other values are being set correctly, like `LEPTOS_SITE_ADDR` seems to be picked up with the value set to `site-addr` in my `Cargo.toml`.
